src/main/kotlin/com/toasttab/pulseman/state/AuthSelector.kt

```diff
@@ -47,30 +51,34 @@ class AuthSelector(
         initialSettings?.selectedAuthClass?.let { savedSelection ->
             selectedAuthClass.selected = authJars.loadedClasses.getClass(savedSelection)
         }
-
-        Theme.load(
-            javaClass.getResourceAsStream(
-                "/org/fife/ui/rsyntaxtextarea/themes/dark.xml"
-            )
-        ).apply { apply(textArea) }
     }
 
-    val sp = RTextScrollPane(textArea)
-
-    fun onFilterChange(newValue: String) {
-        filter.value = newValue
-    }
+    private val sp = RTextScrollPane(textArea)
 
-    fun onSelectedAuthClass(newValue: PulsarAuthHandler) {
+    private fun onSelectedAuthClass(newValue: PulsarAuthHandler) {
         selectedAuthClass.selected =
             if (selectedAuthClass.selected == newValue)
                 null
             else newValue
-        setUserFeedback("Selected ${newValue.cls.name}")
+        setUserFeedback("$SELECTED ${newValue.cls.name}")
         onChange()
     }
 
-    fun filteredClasses() = authJars.loadedClasses.filter(filter.value)
+    private fun filteredClasses() = authJars.loadedClasses.filter(filter.value)
 
     fun authJsonParameters(): String = textArea.text
+
+    @ExperimentalFoundationApi
+    fun getUI(): @Composable () -> Unit {
+        return {
+            authSelectorUI(
+                filter.value,
+                filter::onStateChange,
+                filteredClasses(),
+                selectedAuthClass.selected,
+                ::onSelectedAuthClass,
+                sp
+            )
+        }
+    }
```

Comment:
I've removed as much state as I could from the views, this is a recommended pattern for jetpack compose
https://developer.android.com/jetpack/compose/state#state-hoisting

In general I've encapsulated a lot of these calls into getUI functions. And I've pushed the state as far up the call hierarchy as I could.
I'm sure I can improve further on this.

A lot of the changes to the views below are just pulling the state out of the class and passing it as a primitive variable with a related onChange call.